UnitWorld_shared/shared/game/GameManager.h

```diff
@@ -20,43 +22,49 @@ namespace uw
             _nextAddPlayer(nullptr)
         {}
 
-        ~GameManager()
-        {
-            stop();
-        }
-
         void stop()
         {
             _isRunning = false;
+
+            _physicsThread.join();
         }
 
-        void startSync()
+        void startAsync()
         {
-            while(_isRunning) {
+            _physicsThread = std::thread([this] { loopPhysics(); });
+        }
 
-                const auto startFrameTime = std::chrono::steady_clock::now();
+        void setNextPlayer(std::shared_ptr<Player> newPlayer)
```

Comment:
The name is not telling the full intent. Are we focusing on a particular player, and if so, why?

Reply:
Yes we are here, this adds the next player into the game manager, and is currently used only at a client connection time.

Given this, I don't really get your comment

Reply:
It's up to you, but I though maybe giving it a more descriptive name?

Reply:
Like, for instance, `addPlayer`, `addNextPlayer`, `addNewPlayer`

Reply:
I see. I like the `setNext` because it tells the caller that this will not be taken in account immediately, but rather in the next physics loop. But I also like the `addPlayer` since this is what will be resulting out of this call in the end.

Why not `setNextAddedPlayer`?

Reply:
agreed